modules/notification/webhook/webhook.go

```diff
@@ -769,12 +769,12 @@ func sendReleaseHook(doer *user_model.User, rel *models.Release, action api.Hook
 		return
 	}
 
-	mode, _ := models.AccessLevel(rel.Publisher, rel.Repo)
+	mode, _ := models.AccessLevel(doer, rel.Repo)
```

Comment:
@wxiaoguang This is a bug - the rel.Publisher doesn't get changed between "edits" and "delete" of the release:

```go
func (m *webhookNotifier) NotifyNewRelease(rel *models.Release) {
	sendReleaseHook(rel.Publisher, rel, api.HookReleasePublished)
}

func (m *webhookNotifier) NotifyUpdateRelease(doer *user_model.User, rel *models.Release) {
	sendReleaseHook(doer, rel, api.HookReleaseUpdated)
}

func (m *webhookNotifier) NotifyDeleteRelease(doer *user_model.User, rel *models.Release) {
	sendReleaseHook(doer, rel, api.HookReleaseDeleted)
}
```

Guess this will deserve it's own PR?

Reply:
#17806